lib/components/parallax_component.dart

```diff
@@ -71,52 +73,52 @@ class ParallaxLayer {
     _scale = scale(parallaxImage.fill);
 
     // The image size so that it fulfills the LayerFill parameter
-    _imageSize = Offset(_image.width / _scale, _image.height / _scale);
+    _imageSize =
+        Vector2(_image.width.toDouble(), _image.height.toDouble()) / _scale;
 
     // Number of images that can fit on the canvas plus one
     // to have something to scroll to without leaving canvas empty
-    final countX = 1 + size.width / _imageSize.dx;
-    final countY = 1 + size.height / _imageSize.dy;
+    final countX = 1 + size.x / _imageSize.x;
+    final countY = 1 + size.y / _imageSize.y;
 
     // Percentage of the image size that will overflow
-    final overflowX = (_imageSize.dx * countX - size.width) / _imageSize.dx;
-    final overflowY = (_imageSize.dy * countY - size.height) / _imageSize.dy;
+    final overflowX = (_imageSize.x * countX - size.x) / _imageSize.x;
+    final overflowY = (_imageSize.y * countY - size.y) / _imageSize.y;
 
     // Align image to correct side of the screen
     final alignment = parallaxImage.alignment;
     final marginX = alignment.x == 0 ? overflowX / 2 : alignment.x;
     final marginY = alignment.y == 0 ? overflowY / 2 : alignment.y;
-    _scroll ??= Offset(marginX, marginY);
+    _scroll ??= Vector2(marginX, marginY);
 
-    // Size of the area to paint the images in
-    final rectWidth = countX * _imageSize.dx;
-    final rectHeight = countY * _imageSize.dy;
-    _paintArea = Rect.fromLTWH(0, 0, rectWidth, rectHeight);
+    // Size of the area to paint the images on
+    final paintSize = Vector2(countX, countY)..multiply(_imageSize);
+    _paintArea = paintSize.toOrigoRect();
   }
 
-  void update(Offset delta) {
+  void update(Vector2 delta) {
     if (!loaded()) {
       return;
     }
 
     // Scale the delta so that images that are larger don't scroll faster
-    _scroll += delta.scale(1 / _imageSize.dx, 1 / _imageSize.dy);
+    _scroll += delta..divide(_imageSize);
     switch (parallaxImage.repeat) {
       case ImageRepeat.repeat:
-        _scroll = _scroll % 1;
+        _scroll = Vector2(_scroll.x % 1, _scroll.y % 1);
         break;
       case ImageRepeat.repeatX:
-        _scroll = Offset(_scroll.dx % 1, _scroll.dy);
+        _scroll = Vector2(_scroll.x % 1, _scroll.y);
         break;
       case ImageRepeat.repeatY:
-        _scroll = Offset(_scroll.dx, _scroll.dy % 1);
+        _scroll = Vector2(_scroll.x, _scroll.y % 1);
         break;
       case ImageRepeat.noRepeat:
         break;
     }
 
-    final dx = _scroll.dx * _imageSize.dx;
-    final dy = _scroll.dy * _imageSize.dy;
+    final dx = _scroll.x * _imageSize.x;
+    final dy = _scroll.y * _imageSize.y;
 
     _paintArea = Rect.fromLTWH(-dx, -dy, _paintArea.width, _paintArea.height);
```

Comment:
we should definitely add methods to make rects, like combine one vector for p and one vector for size to get a rect, and a version in witch the p is zero. but that can be a followup.

Reply:
Already exists in the extension:
```
  /// Creates a [Rect] starting from [x, y] and going the [Vector2]
  Rect toRect(Vector2 to) => Rect.fromLTWH(x, y, to.x, to.y);

  /// Creates a [Rect] starting in origo and going the [Vector2]
  Rect toOrigoRect() => Rect.fromLTWH(0, 0, x, y);
```

Reply:
Moved to `RectFactory`